hibernate-core/src/main/java/org/hibernate/proxy/pojo/javassist/JavassistLazyInitializer.java

```diff
@@ -57,6 +57,11 @@ public Object invoke(
 			final Method proceed,
 			final Object[] args) throws Throwable {
 		if ( this.constructed ) {
```

Comment:
Ah, OK, I missed that the methods had the same name but were for different classes. Makes sense to me now.

Regarding my first comment, I understand you are avoiding initialization due to calling a method prefixed `...$$_hibernate_getEntityEntry`, which is the desired behavior.

This PR looks fine to me.